src/main/java/com/sleekbyte/tailor/Tailor.java

```diff
@@ -95,27 +145,38 @@ public static void exitWithNoSourceFilesError() {
                 }
 
             } catch (ReflectiveOperationException e) {
-                throw new CliArgumentParserException("Listeners were not successfully created: " + e);
+                throw new CLIArgumentParserException("Listeners were not successfully created: " + e);
             }
         }
+
+        listeners.add(new MinLengthListener(printer, constructLengths, enabledRules));
+        listeners.add(new MaxLengthListener(printer, constructLengths, enabledRules));
+        DeclarationListener decListener = new DeclarationListener(listeners);
+        listeners.add(decListener);
+
         return listeners;
     }
 
     /** Runs SwiftLexer on input file to generate token stream.
      *
-     * @param inputFile Lexer input
+     * @param input Lexer input
      * @return Token stream
-     * @throws IOException if file cannot be opened
-     * @throws CliArgumentParserException if cmd line arguments cannot be parsed
      */
-    public static CommonTokenStream getTokenStream(File inputFile) throws IOException, CliArgumentParserException {
-        FileInputStream inputStream = new FileInputStream(inputFile);
-        SwiftLexer lexer = new SwiftLexer(new ANTLRInputStream(inputStream));
-        if (!configuration.debugFlagSet()) {
-            lexer.removeErrorListeners();
-            lexer.addErrorListener(new ErrorListener());
+    private Optional<CommonTokenStream> getTokenStream(File input) {
+        try {
+            FileInputStream inputStream = new FileInputStream(input);
+            SwiftLexer lexer = new SwiftLexer(new ANTLRInputStream(inputStream));
+            if (!configuration.debugFlagSet()) {
+                lexer.removeErrorListeners();
+                lexer.addErrorListener(new ErrorListener());
+            }
+            return Optional.of(new CommonTokenStream(lexer));
+        } catch (IOException e) {
+            handleIOException(e);
+        } catch (CLIArgumentParserException e) {
+            handleCLIException(e, configuration);
         }
-        return new CommonTokenStream(lexer);
+        return Optional.empty();
```

Comment:
No (because of the `System.exit()` calls in the `handle…` methods), but the method requires a return statement if the control flow skips the first return due to an exception.